src/extension-runners/firefox-android.js

```diff
@@ -411,12 +411,10 @@ export class FirefoxAndroidExtensionRunner {
     log.debug('Checking read/write permissions needed for web-ext' +
               `on ${selectedFirefoxApk}...`);
 
-    // Runtime permission needed to be able to run Firefox on a temporarily created profile
-    // on android versions >= 23 (Android Marshmallow, which is the first version where
-    // these permissions are optional and have to be granted explicitly).
+    // Runtime permissions needed to Firefox to be able to access the
+    // xpi file uploaded to the android device or emulator.
     const requiredPermissions = [
       'android.permission.READ_EXTERNAL_STORAGE',
```

Comment:
I just tested, and the read permission isn't even needed to run an extension in Fenix. We can delete lots of code.

Reply:
@Rob--W have you checked that the permission wasn't actually already granted?
some more details about what STR you actually tested manually would be helpful to assess if that was enough to ensure we are not missing some legit corner case.

Reply:
@Rob--W also, I would like to release this into a minor or patch version, and so "less changes" would make me feel more less concerned than "remove a lot of code" about releasing it in a "non major" new release.

I would be obviously more than happy about "remove a lot of code" in a separate follow up.

Reply:
Filed https://github.com/mozilla/web-ext/issues/2312